src/runtime/CMakeLists.txt

```diff
@@ -217,17 +218,17 @@ foreach (i IN LISTS RUNTIME_CPP)
                     # unfortunately, clang doesn't automatically set this flag even though the
                     # ABI is msvc on windows
                     set(fshort-wchar -fshort-wchar)
```

Comment:
The comments here say the ABI is `msvc`... is that accurate? What would happen if we used `x86_64-unknown-windows-msvc` as the target? Would we still need the `-fshort-wchar` flag in that case?

Reply:
I have no idea, that's a question for @slomp

Reply:
Yup, we need to adhere to the MSVC ABI because D3D12 is shipped as a DLL, and we need to talk to it with the "standard" Windows x64 ABI rules.
A long time ago, when d3d12 back-end shipped, I ended up implementing llvm assembly stubs to patch calls to D3D12 DLL, but in this PR https://github.com/halide/Halide/pull/5217 we decided that changing the ABI in the triplet was the right thing to do.

> Would we still need the -fshort-wchar flag in that case?
No idea. The `-fshort-wchar` documentation is abysmal, and non-MSVC compiler support for Windows shenanigans is janky at best.

Reply:
> and non-MSVC compiler support for Windows shenanigans is janky at best.

IIRC we removed MinGW support ages ago and no one has asked for it back c.f. #4601

Reply:
I'm not sure when we started setting that flag, exactly, but `sizeof(wchar_t)` has been `2` when the OS target is Windows since Clang 3.5

https://gcc.godbolt.org/z/3rWP7r8qT